### PR TITLE
[WIP] Added functionality for fetching commands

### PR DESCRIPTION
**Pull request by SamAlexandrov**
Still work in progress, it will be tested in Sat 8 July 2017.

### CODE REVIEW

File: challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+def fetch(robot, time_out=15.0):
+
+    confirmed = False
+    robot.speech.speak('What can I do for you, master?')
+
+    # Listen for the new task
+    while not confirmed:
+        try:
+            sentence, semantics = robot.hmi.query('', 'T -> see the taa | other open', 'T')
```
Comment (LoyVanBeek) [inline challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py]: taa?

File: challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    
+            # check if we have heard this correctly
+            robot.speech.speak('I heard %s, is this correct?' % sentence)
+            try:
+                if 'yes' == robot.hmi.query('', 'T -> yes | no', 'T').sentence:
```
Comment (LoyVanBeek) [inline challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py]: The order in this comparison is a bit odd, usually, the constant comes last

File: challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+from robot_smach_states import Initialize, Say
+from robot_smach_states.util.startup import startup
+
+def fetch(robot, time_out=15.0):
+
+    confirmed = False
+    robot.speech.speak('What can I do for you, master?')
+
+    # Listen for the new task
+    while not confirmed:
```
Comment (LoyVanBeek) [inline challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py]: This loop can now be infinite? Please put a reasonable max on it with a proper alternative

File: challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        def __init__(self, robot, time_out=15.0):
+                smach.State.__init__(self, outcomes=["heard"])
+                self.robot = robot
+                self.time_out = time_out
+
+        def execute(self, userdata):
+                
+                self.robot.head.look_at_standing_person()
+
+                fetch_command = fetch(self.robot, time_out=self.time_out)
```
Comment (LoyVanBeek) [inline challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py]: Why not inline this function? Just pu the loop inplementation here directly or as a class method as well?

File: challenge_set_a_table/src/challenge_set_a_table.py [hunk 5]
```diff
@@ -25,38 +30,73 @@ def __init__(self, robot):
             #Part I: Set a table
             smach.StateMachine.add('INITIALIZE',
                                    Initialize(robot),
```
Comment (MatthijsBurgh) [inline challenge_set_a_table/src/challenge_set_a_table.py]: This is done in StartChallengeRobust, so you should remove this

File: challenge_set_a_table/src/challenge_set_a_table.py [hunk 6]
```diff
@@ -1,20 +1,25 @@
 #! /usr/bin/env python
 
+# ------------------------------------------------------------------------------------------------------------------------
```
Comment (MatthijsBurgh) [inline challenge_set_a_table/src/challenge_set_a_table.py]: Please remove this, not done in any challenge

File: challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+from robot_smach_states import Initialize, Say
+from robot_smach_states.util.startup import startup
+
+def fetch(robot, time_out=15.0):
+
+    confirmed = False
+    robot.speech.speak('What can I do for you, master?')
+
+    # Listen for the new task
+    i = 0
```
Comment (MatthijsBurgh) [inline challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py]: use self explanatory variable names

Comment (LoyVanBeek) [inline challenge_set_a_table/src/challenge_set_a_table_states/fetch_command.py]: For iterations, ```i``` is quite accepted actually.

**Comment by LoyVanBeek** [comment]
I merged this too soon. @SamAlexandrov knowledge is still missing.

**Pull request merged**